### PR TITLE
Add pyre2 package

### PR DESCRIPTION
**Pull request by sarnold**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sarnold** [comment]
Since this is a cython extension and I can't ping help-python-c directly =>  [ping the team](https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team)

File: recipes/pyre2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{% set name = "pyre2" %}
+{% set version = "0.3.2" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
```
Comment (ocefpaf) [inline recipes/pyre2/meta.yaml:6]: I believe that those quotes jinja variables will break the updating bot.

File: recipes/pyre2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: true  # [win]
+  skip: true  # [py<36]
+
+requirements:
+  build:
+    - "{{ compiler('cxx') }}"
+    - cython
```
Comment (ocefpaf) [inline recipes/pyre2/meta.yaml:21]: You probably don't need cython here, on in host should be enough.

File: recipes/pyre2/meta.yaml (hunk 1 above)
Comment (sarnold) [reply recipes/pyre2/meta.yaml:6]: Hmm, I took that directly from the docs (changing now).

File: recipes/pyre2/meta.yaml (hunk 2 above)
Comment (sarnold) [reply recipes/pyre2/meta.yaml]: Lemme test again in a conda env, but I think that's required at build time (since the .cpp module src is not packaged).

Comment (ocefpaf) [reply recipes/pyre2/meta.yaml]: Host is available at build time and make it cross-compile-able. We only put things in build that cannot be in host.

Comment (sarnold) [reply recipes/pyre2/meta.yaml]: Works fine in fresh conda-forge env (cython not required under build).  The next version will most likely have cmake and pybind11 deps.  Thanks!

**Pull request merged**